src/app/InteractionModelTimeout.h

```diff
@@ -18,11 +18,15 @@
 #pragma once
 
 #include <system/SystemClock.h>
+#include <transport/Session.h>
+#include <transport/SessionHandle.h>
 
 namespace chip {
 namespace app {
 
-static constexpr System::Clock::Timeout kImMessageTimeout = System::Clock::Seconds16(12);
+// This is an expected upper bound of time-to-first-byte for IM transactions, the actual processing time should never exceed this
+// value unless specified elsewhere (e.g. async command handling for some clusters).
```

Comment:
This should be a lot less than 5s no, since in almost all cases, the processing is not IO bound, nor is it application dependent? I'd think it should be around 2s, max.

This is a default. If specific IM uses need higher, it can be set appropriately at the call site.